src/Provider/GeocodeEarth/Tests/GeocodeEarthTest.php

```diff
@@ -0,0 +1,287 @@
+    }
+
+    public function testGetName()
+    {
+        $provider = new GeocodeEarth($this->getMockedHttpClient(), 'api_key');
+        $this->assertEquals('geocode_earth', $provider->getName());
+    }
+
+    public function testGeocode()
+    {
+        $provider = new GeocodeEarth($this->getMockedHttpClient('{}'), 'api_key');
+        $result = $provider->geocodeQuery(GeocodeQuery::create('foobar'));
+
+        $this->assertInstanceOf(Collection::class, $result);
+        $this->assertEquals(0, $result->count());
+    }
+
+    public function testGeocodeWithRealAddress()
+    {
+        if (!isset($_SERVER['GEOCODE_EARTH_API_KEY'])) {
+            $this->markTestSkipped('You need to configure the GEOCODE_EARTH_API_KEY value in phpunit.xml');
+        }
+
+        $provider = new GeocodeEarth($this->getHttpClient($_SERVER['GEOCODE_EARTH_API_KEY']), $_SERVER['GEOCODE_EARTH_API_KEY']);
+        $results = $provider->geocodeQuery(GeocodeQuery::create('242 Acklam Road, London, United Kingdom'));
+
+        $this->assertInstanceOf('Geocoder\Model\AddressCollection', $results);
+        $this->assertCount(1, $results);
+
+        /** @var \Geocoder\Model\Address $result */
+        $result = $results->first();
+        $this->assertInstanceOf('\Geocoder\Model\Address', $result);
+        $this->assertEquals(51.521124, $result->getCoordinates()->getLatitude(), '', 0.01);
+        $this->assertEquals(-0.20360200000000001, $result->getCoordinates()->getLongitude(), '', 0.01);
+        $this->assertEquals('Acklam Road', $result->getStreetName());
+        $this->assertEquals('London', $result->getLocality());
+        $this->assertCount(3, $result->getAdminLevels());
+        $this->assertEquals('London', $result->getAdminLevels()->get(3)->getName());
+        $this->assertEquals('United Kingdom', $result->getCountry()->getName());
+        $this->assertEquals('GBR', $result->getCountry()->getCode());
+    }
+
+    public function testReverseWithRealCoordinates()
+    {
+        if (!isset($_SERVER['GEOCODE_EARTH_API_KEY'])) {
+            $this->markTestSkipped('You need to configure the GEOCODE_EARTH_API_KEY value in phpunit.xml');
+        }
+
+        $provider = new GeocodeEarth($this->getHttpClient($_SERVER['GEOCODE_EARTH_API_KEY']), $_SERVER['GEOCODE_EARTH_API_KEY']);
+        $results = $provider->reverseQuery(ReverseQuery::fromCoordinates(54.0484068, -2.7990345));
+
+        $this->assertInstanceOf('Geocoder\Model\AddressCollection', $results);
+        $this->assertCount(5, $results);
+
+        /** @var \Geocoder\Model\Address $result */
+        $result = $results->first();
+        $this->assertInstanceOf('\Geocoder\Model\Address', $result);
+        $this->assertEquals(54.048411999999999, $result->getCoordinates()->getLatitude(), '', 0.001);
+        $this->assertEquals(-2.7989549999999999, $result->getCoordinates()->getLongitude(), '', 0.001);
+        $this->assertEquals(1, $result->getStreetNumber());
```

Comment:
I've updated these tests to match responses from this new provider. Just curious if there are any guidelines for writing tests?

Reply:
No really, as long as you test "everything" it's a good test :)